src/v2/Components/NavBar/NavBar.tsx

```diff
@@ -149,7 +150,7 @@ export const NavBar: React.FC = track(
                   flex={1}
                   size="small"
                 >
-                  Sign Up
+                  <FormattedMessage id="navbar.signup" />
```

Comment:
Is there type support (like do these ids autocomplete)?

Reply:
if this doesnt exist, having a hook or a global func could totally solve this with TS.
```
const t = useTranslations()
return (
  <Text>
    t("navbar.signup")
  </Text>
)
```

Reply:
https://formatjs.io/docs/react-intl/api/

https://formatjs.io/docs/react-intl/components#why-components

It seems like this lib does a lot and tries to be a do-all solution for text formatting. Which 🤷 

I'm not sure why the hook wouldn't "Provide TypeScript type definitions" though?

Reply:
m indeed. we can at least do what we do with palette tokens, and wrap their func in our func, and add types based on our json translation files 🤔